static/src/javascripts/projects/common/modules/identity/ad-prefs.js

```diff
@@ -0,0 +1,123 @@
+// @flow
+
+import React, { Component } from 'react';
+import { render } from 'react-dom';
+import fastdom from 'lib/fastdom-promise';
+import {
+    getAdConsentState,
+    setAdConsentState,
+    allAdConsents,
+} from './ad-prefs.lib';
+import type { AdConsent } from './ad-prefs.lib';
+
+const rootSelector: string = '.js-manage-account__ad-prefs';
+
+type ConsentRadioButtonProps = {
+    value: string,
+    label: string,
+    checked: boolean,
+    consent: AdConsent,
+    onToggle: () => void,
+};
+
+class ConsentRadioButton extends Component<ConsentRadioButtonProps, {}> {
+    handleChange(event): void {
+        if (event.target.checked) {
+            this.props.onToggle();
+        }
+    }
+    render() {
+        const id = `gu-ad-prefs-${this.props.value.toString()}-${
+            this.props.consent.cookie
+        }`;
+        const name = `gu-ad-prefs-${this.props.consent.cookie}`;
+
+        return (
+            <div>
+                <label htmlFor={id}>
+                    <input
+                        type="radio"
+                        name={name}
+                        id={id}
+                        value={this.props.value.toString()}
+                        checked={this.props.checked}
+                        onChange={this.handleChange.bind(this)}
+                    />
+                    {this.props.label}
+                </label>
+            </div>
+        );
+    }
+}
+
+class ConsentBox extends Component<
+    { consent: AdConsent },
+    { consentState: ?boolean }
+> {
+    constructor(props) {
```

Comment:
Could you add a type to `props`?

Reply:
they are already defined above in the component type definition, should I duplicate them, extract them to a variable, remove from the component?

Reply:
Ok yeah you're good 👍